File: APIdoc/java/pio/src/main/java/io/point/tutorials/ListFoldersCheckInCheckOut.java
package io.point.tutorials;

import io.point.Pio;
import org.codehaus.jackson.JsonNode;
import org.codehaus.jackson.node.ArrayNode;

import java.util.Iterator;

/**
 * User: jconley
 * Date: 5/6/2014
 */
public class ListFoldersCheckInCheckOut {

    public static void main(String[] args) throws Exception{
        String email = "";
        String password = "";
        String apiKey = "";

        String sessionKey = authenticate(email, password, apiKey);

        ArrayNode accessRulesArray = Pio.listAccessRules(sessionKey);
        JsonNode firstRule = accessRulesArray.get(0);

        ArrayNode folderArray = Pio.listFolders(sessionKey, firstRule.get(1).getTextValue());
        Iterator<JsonNode> rootObjects = folderArray.getElements();
        while(rootObjects.hasNext()) {
            ArrayNode file = (ArrayNode)rootObjects.next();

            //if it's a director, create a link for it and exit
            if(file.get(2).getTextValue().equals("FILE")){
                Pio.checkout(sessionKey, file.get(16).asText(), file.get(0).asText(), file.get(1).asText());

                String downloadUrl = Pio.fileDownload(sessionKey, file.get(16).asText(), file.get(0).asText(), file.get(1).asText());

                //follow link and download file

                //edit file
                String localPath = "";

                Pio.fileUpload(sessionKey, file.get(16).asText(), file.get(0).asText(), file.get(1).asText(), localPath);

                Pio.checkin(sessionKey, file.get(16).asText(), file.get(0).asText(), file.get(1).asText());

                break;
            }
        }
    }
}

File: APIdoc/php/tutorials/listFoldersCheckInCheckOut.php
$email = "";
$password = "";
$apiKey = "";
$sessionKey = auth($email, $password, $apiKey);

$rules = list_access_rules($sessionKey)["RESULT"]["DATA"];
$folders = $list_folders($sessionKey, $rules[2]);
foreach($folders as $f){	
	if($f[2] == "FILE"){
		$checkout = file_checkout(sessionKey, $f[16], $f[0], $f[1]);

		$download = file_download(sessionKey, $f[16], $f[0], $f[1]);

		##follow download link, download file and make edits, load new file into contents
		$contents = "";

		$upload = file_upload(sessionKey, $f[16], $f[0], $f[1], $contents);

		$checkin = file_checkin(sessionKey, $f[16], $f[0], $f[1]);
		echo $checkin;
	}
}